server/src/ws/app_handler/methods/initialize_session.rs

```diff
@@ -79,15 +79,22 @@ pub async fn initialize_session_connection(
         }
     };
 
-    let app_sessions_read = sessions_write
-        .get(app_id)
-        .expect("Session just created or updated; unwrap safe")
-        .read()
-        .await;
+    let app = match sessions_write.get(app_id) {
```

Comment:
instead of doing this pass data to response from above when we access or create new session, this way we won't have to access the same data twice